src/lib/components/remark/urls.js

```diff
@@ -0,0 +1,44 @@
+import { findAndReplace } from "mdast-util-find-and-replace";
+
+// scheme://authority/path?query#fragment
+const urlRegex = /(([^:\/?#\s]+):\/\/(?:([^\/?#\s]*))?([^?\s]*)(?:\?([^#\s]*))?(?:#(\S*))?)/gi;
+
+export default function urls() {
+  function replace(value, url, scheme, authority, path, query, fragment, match) {
+    if (
+      /[\w`]/.test(match.input.charAt(match.index - 1)) ||
+      /[/\w`]/.test(match.input.charAt(match.index + value.length))
+    ) {
+      return false;
+    }
+
+    // parse query with native URLSearchParams
+    query = query && Object.fromEntries(new URLSearchParams(query));
+
+    let node = { type: "text", value };
```

Comment:
Is this assignment to `node` necessary? I also find this in hashtag and mention.